flytekit/extras/tasks/shell.py

```diff
@@ -204,6 +218,11 @@ def execute(self, **kwargs) -> typing.Any:
         if os.name == "nt":
             self._script = self._script.lstrip().rstrip().replace("\n", "&&")
 
+        if "env" in kwargs and isinstance(kwargs["env"], dict):
```

Comment:
how about extending ShellTask to make a new tasktype called RawShellTask?
this could automaticaly export all vars as env?
And the helper method that you have can be added to that.
This way, there are no side-effects, like un-intended use of env etc and keeps shell task exactly what it is today. 

Also cc @palchicz for comments on this?

Reply:
Yes, I had originally considered this but I had two reservations:

1. The desired behavior I wanted to add is so minimal that stamping out a new class entirely, only to copy 95% of the `execute()` methods felt overly verbose.
2. The main issue of setting arbitrary environment variables at execution time is still not easily addressed. 

I am open to your suggestion but could you provide a more concrete plan for implementation? I supposed I could re-work the class in such a way that the intended "raw" script be passed in via `script_file` but I still need the `script_args` and `env` variables to be passed in at execution time, and there is no easy way to add additional params to `execute()` since it goes through the `flyte_entity_call_handler` and all `inputs` are eventually exposed via the `kwargs` in `execute()`. So I could either accept `script_args` and `env` as "special" inputs, or accept no other inputs but these two and limit the use-case. The class would still need to be instantiated with a the generic template wrapper (likely provided by the class with ability to override). I'm happy to discuss this further, my solution here was intended to be as minimal since I am not familiar with the patterns and customs of this code base

Reply:
I'm strongly in favor of creating a new class instead of modifying the current one (i.e. favor extension over modification), especially since this would change the API of the class and it is already in use. 

It would also have a positive impact on the code base if the internals of this class were reworked such that new but slightly different classes are easier to stamp out. No real suggestions on implementation to get there though :-/